Libplanet/Net/Protocols/KademliaProtocol.cs

```diff
@@ -0,0 +1,525 @@
+                }
+                catch (TimeoutException)
+                {
+                    _logger.Error("A timeout exception occurred connecting to seed peer.");
+                    continue;
+                }
+                catch (Exception)
+                {
+                    _logger.Error("An unexpected exception occurred connecting to seed peer.");
+                    continue;
+                }
+            }
+
+            if (findPeerTasks.Count == 0)
+            {
+                throw new SwarmException("Bootstrap failed.");
+            }
+
+            try
+            {
+                await Task.WhenAll(findPeerTasks);
+            }
+            catch (TimeoutException)
+            {
+                bool timeoutOccurred = findPeerTasks.All(findPeerTask => findPeerTask.IsFaulted);
+
+                if (timeoutOccurred)
+                {
+                    throw new TimeoutException(
+                        "Timeout exception occurred during BootstrapAsync().");
+                }
+            }
+            catch (Exception)
+            {
+                _logger.Error("An unexpected exception occurred during BootstrapAsync().");
+                throw;
+            }
+        }
+
+        public async Task RefreshAsync(CancellationToken cancellationToken)
+        {
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                await Task.Delay(IdleBucketRefreshInterval, cancellationToken);
+
+                byte[] buffer = new byte[20];
+                List<Task> tasks = new List<Task>();
+                for (int i = 0; i < FindConcurrency; i++)
+                {
+                    _random.NextBytes(buffer);
+                    tasks.Add(FindPeerAsync(
+                        new Address(buffer),
+                        null,
+                        RequestTimeout,
+                        cancellationToken));
+                }
+
+                tasks.Add(FindPeerAsync(_address, null, RequestTimeout, cancellationToken));
+                try
+                {
+                    await Task.WhenAll(tasks);
+                }
+                catch (TimeoutException)
+                {
+                }
+            }
+        }
+
+#pragma warning disable CS4014 // 이 호출을 대기하지 않으므로 호출이 완료되기 전에 현재 메서드가 계속 실행됩니다.
+        public void ReceiveMessage(object sender, Message message)
+        {
+            switch (message)
+            {
+                case Ping ping:
+                    ReceivePingAsync(ping);
+                    break;
+
+                case FindPeer findPeer:
+                    ReceiveFindPeerAsync(findPeer);
+                    break;
+
+                default:
+                    UpdateAsync(message.Remote, _cancellationToken);
+                    break;
+            }
+        }
+#pragma warning restore CS4014 // 이 호출을 대기하지 않으므로 호출이 완료되기 전에 현재 메서드가 계속 실행됩니다.
+
+        public string Trace()
```

Comment:
What purpose of this method?

Reply:
A test method to investigate the routing table of the peer.